Flag to build against http-client-openssl

mmhat opened the pull request:
Enabling this flag will build the library against `http-client-openssl` instead of `http-client-tls`.

arybczak commented:
I'd be inclined to use `http-client-openssl` by default and drop the `tls` version, `openssl` seems like a safer bet.

But from looking at https://hackage.haskell.org/package/http-client-openssl-0.3.3/docs/Network-HTTP-Client-OpenSSL.html it looks like a call to withOpenSSL is missing?

mmhat commented:
> I'd be inclined to use `http-client-openssl` by default and drop the `tls` version, `openssl` seems like a safer bet.

I think that would be a good idea. I changed the default of the `openssl` flag but I left the option to use `tls` (for now).
I would not mind to drop `tls` support entirely; I am happy as long as I can use this library with OpenSSL.

> looks like a call to withOpenSSL is missing?

@arybczak You are absolutely right; I fixed that. Thank you!

arybczak commented:
Can you rebase and fix the build?

mmhat commented:
Done.

mmhat commented:
Are there any issues left that prevent merging? If so, please let me know and I am happy to address those.

arybczak commented:
Sorry, that completely fell off my radar. It's good, thanks :slightly_smiling_face:
